### PR TITLE
Use vitest

### PR DESCRIPTION
**Pull request by wonkyDD**

### CODE REVIEW

File: .all-contributorsrc
Comment (PatrykKuniczak) [inline .all-contributorsrc:1]: I think good idea is disable prettier for this file, cause this bot add new contributor to this file automattically and don't reformat like prettier.

File: README.md
Comment (PatrykKuniczak) [inline README.md:1]: IMO the previouse version is better, cause have short and clear links, paste entire message as a link, isn't good idea.

Create here only the things related to your feature :)

File: src/pages/devtools/index.html
Comment (PatrykKuniczak) [inline src/pages/devtools/index.html:1]: i think it's bug with prettier, but i think why you run prettier for entire project, i was configure lint-staged, for run prettier and eslint only for staged file.

File: package.json [hunk 1]
```diff
@@ -36,14 +36,13 @@
...
     "@types/chrome": "0.0.248",
-    "@types/jest": "29.5.7",
     "@types/node": "20.8.10",
     "@types/react": "18.2.35",
     "@types/react-dom": "18.2.14",
     "@types/ws": "8.5.8",
     "@typescript-eslint/eslint-plugin": "6.9.1",
     "@typescript-eslint/parser": "6.9.1",
-    "@vitejs/plugin-react": "2.2.0",
+    "@vitejs/plugin-react": "4.1.1",
```
Comment (PatrykKuniczak) [inline package.json:45]: You can't update this read more in https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/258#issuecomment-1794608629

File: package.json [hunk 2]
```diff
@@ -56,18 +55,17 @@
...
     "npm-run-all": "4.1.5",
     "prettier": "3.0.3",
     "rollup": "4.3.0",
     "sass": "1.69.5",
-    "ts-jest": "29.1.1",
     "ts-loader": "9.5.0",
     "tslib": "2.6.2",
     "typescript": "5.2.2",
-    "vite": "3.2.7",
+    "vite": "4.5.0",
```
Comment (PatrykKuniczak) [inline package.json:67]: You can't update this read more in https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/258#issuecomment-1794608629

File: package.json [hunk 3]
```diff
@@ -56,18 +55,17 @@
...
+    "vitest": "^0.34.6",
```
Comment (PatrykKuniczak) [inline package.json:68]: I think the good approach is don't use '^' anyway, but here we have deps bot, and they are looking for new versions, as you can see in other packages we don't have it, but here @Jonghakseo could say more if he want this, let's do it :)

File: package.json [hunk 4]
```diff
@@ -56,18 +55,17 @@
     "eslint-plugin-react-hooks": "4.6.0",
     "fs-extra": "11.1.1",
     "husky": "8.0.3",
-    "jest": "29.7.0",
-    "jest-environment-jsdom": "29.7.0",
+    "jsdom": "^22.1.0",
```
Comment (PatrykKuniczak) [inline package.json:58]: I think the good approach is don't use '^' anyway, but here we have deps bot, and they are looking for new versions, as you can see in other packages we don't have it, but here @Jonghakseo could say more if he want this, let's do it :)

**Review by PatrykKuniczak — CHANGES_REQUESTED** [review]

**Comment by PatrykKuniczak** [comment]
Don't panic 😁

Let's open new one

**Comment by wonkyDD** [comment]
@PatrykKuniczak 
I should have rebased the upstream commit beforehand and fixed the current branch, but I got screwed.
Thanks for the feedback.

> You can't update this read more in [#258 (comment)](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/258#issuecomment-1794608629)

To use vitest successfully, we need to upgrade `@vitejs/plugin-react` and `vite`. 
If not, it will encounter this issue https://github.com/vitest-dev/vitest/issues/3283 and tests will continue to fail.

However, after reading the issue you linked to, I realized that I am not able to update it at this time.
I should have read the description of the package update PR beforehand, but I didn't. (apologize for bothering you 🙇)

As you said, I think we should also hold off on [using vitest](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/issues/257) until the issue is resolved or
there is a workaround.

Thanks a lot.

**Comment by PatrykKuniczak** [comment]
@wonkyDD You no bothered me  😁
If you have small experience with git don't push force, if you no confident your changes, cause you can destroy you work, create also backup branch before rebase, to hqve possibility to recover your work 👻

I was thinking about workaround, i can equal the source of the file, but there's performance on larger files could be sick.

If i have more time on this month, i can try to solve the PR on rollup, then they can relase new version, and then we can pull this and update, as you can see, everything is blocked 😆

**Pull request closed without merging**